corehq/apps/data_dictionary/views.py

```diff
@@ -238,15 +344,65 @@ def post(self, request, *args, **kwargs):
 def _process_bulk_upload(bulk_file, domain):
     filename = make_temp_file(bulk_file.read(), file_extention_from_filename(bulk_file.name))
     errors = []
+    import_fhir_data = toggles.FHIR_INTEGRATION.enabled(domain)
+    fhir_resource_type_by_case_type = {}
+    expected_columns_in_prop_sheet = 5
+
+    if import_fhir_data:
+        expected_columns_in_prop_sheet = 7
+
     with open_any_workbook(filename) as workbook:
         for worksheet in workbook.worksheets:
+            if worksheet.title == FHIR_RESOURCE_TYPE_MAPPING_SHEET:
+                if import_fhir_data:
+                    _errors, fhir_resource_type_by_case_type = _process_fhir_resource_type_mapping_sheet(
+                        domain, worksheet)
+                    errors.extend(_errors)
+                continue
             case_type = worksheet.title
             for (i, row) in enumerate(itertools.islice(worksheet.iter_rows(), 1, None)):
-                if len(row) < 5:
+                if len(row) < expected_columns_in_prop_sheet:
                     error = _('Not enough columns')
                 else:
-                    name, group, data_type, description, deprecated = [cell.value for cell in row[:5]]
-                    error = save_case_property(name, case_type, domain, data_type, description, group, deprecated)
+                    if import_fhir_data:
+                        name, group, data_type, description, deprecated, fhir_resource_prop_path, remove_path = [
+                            cell.value for cell in row[:7]]
+                        remove_path = remove_path == 'Y' if remove_path else False
+                        fhir_resource_type = fhir_resource_type_by_case_type.get(case_type)
+                        if fhir_resource_prop_path and not fhir_resource_type:
+                            error = _('Could not find resource type for {}').format(case_type)
+                        else:
+                            error = save_case_property(name, case_type, domain, data_type, description, group,
```

Comment:
note to self: can save_case_property be still called just once outside if block?